server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -187,9 +201,26 @@ void execute() {
                 try (var ignored = threadPool.getThreadContext().clearTraceContext()) {
                     fetchIndicesStats();
                 }
+                try (var ignored = threadPool.getThreadContext().clearTraceContext()) {
+                    fetchNodesHeapUsage();
+                }
             }
         }
 
+        private void fetchNodesHeapUsage() {
+            heapUsageSupplier.getClusterHeapUsage(ActionListener.releaseAfter(new ActionListener<>() {
+                @Override
+                public void onResponse(Map<String, HeapUsage> stringHeapUsageMap) {
+                    nodesHeapUsage = stringHeapUsageMap;
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    logger.warn("failed to fetch heap usage for nodes", e);
```

Comment:
I think we should set `nodesHeapUsage` back to `Map.of()` in this case? Perhaps can use a test for the error case as well.

Reply:
Done in 765ade8acb08f6624c9c81d3afb0c1262963b955